### ISSUE TITLE
TaskScheduler doesn't order tasks correctly

### ISSUE THREAD

**Issue by Minus20Five** [se701g2/Doto#274]
### Bug Summary: 

TaskScheduler doesn't rearrange existing tasks when adding a new task with an earlier due date .

--- 

### Steps To Reproduce
1) Add a task A with a later due date (e.g. 3 hours in the future)
2) Add a task B with an earliest due date (e.g. 1 hour in the future)

### Expected behaviour

task B should be ordered before task A.

### Observed Behaviour

task B ordered after task A.

### Environment

- Version: latest
- Operating System:  macOS
- Browser (if any): Chrome

---

**Comment by Minus20Five** [comment]
Note that due dates are not yet persisted on the server.
Blocked by #273 which will probably add it.
